lib/src/intl/number_format.dart

```diff
@@ -523,7 +523,7 @@ class NumberFormat {
   /// Helper to round a number which might not be num.
   dynamic _round(dynamic number) {
     if (number is num) {
-      if (number.isInfinite) {
+      if (number.isInfinite || number.isNaN) {
```

Comment:
This does not seem correct. If the value is NaN, there is no correct formatting, and `number.round()` should be allowed to to throw an Error.